src/Altinn.Profile.Core/OrganizationNotificationAddresses/OrganizationNotificationAddressesService.cs

```diff
@@ -55,13 +61,21 @@ public class OrganizationNotificationAddressesService(IOrganizationNotificationA
             }
 
             var duplicateAddress = org.NotificationAddresses?.FirstOrDefault(x => x.FullAddress == notificationAddress.FullAddress && x.AddressType == notificationAddress.AddressType);
-            if (duplicateAddress != null)
+            if (duplicateAddress is { IsSoftDeleted : not true })
             {
                 return (duplicateAddress, true);
             }
 
             var registryId = await _updateClient.UpdateNotificationAddress(existingNotificationAddress.RegistryID, notificationAddress, organizationNumber);
 
+            if (duplicateAddress is { IsSoftDeleted: true })
+            {
+                var restoredAddress = await _orgRepository.RestoreNotificationAddress(duplicateAddress.NotificationAddressID, registryId);
+                await _orgRepository.DeleteNotificationAddressAsync(existingNotificationAddress.NotificationAddressID);
```

Comment:
Is it impossible for `duplicateAddress` and `existingNotificationAddress` to be the same entry? We never compare them. The logic assumes they're never the same.

Reply:
Interesting question. 
What would be the expected outcome if 
* A user does update on an address with the exact same values as it already is
* A user does update on an soft-deleted address with the exact same values as it already is